source/develop/release-management/features/network/ipv6_subnets.html.md

```diff
@@ -0,0 +1,150 @@
+
+Related patches can be found on [gerrit topic:provider_ipv6_subnets][]https://gerrit.ovirt.org/#/q/topic:provider_ipv6_subnets.
+
+### Owner
+* Feature Owner: [Miguel Duarte Barroso](https://github.com/maiqueb) (mbarroso in #ovirt)
+* Email: <mdbarroso@redhat.com>
+
+### Terminology
+
+* Networking API: [OpenStack RESTful Networking API](https://developer.openstack.org/api-ref/network/v2/)
+* OVN: Open Virtual Network. OVN is an OVS (Open vSwitch) extension that
+brings Software Defined Networking to OVS.
+* oVirt OVN provider: a proxy that implements a subset of the Networking API
+and interacts with OVN.
+* subnet: a layer 3 concept, created on top of a network. As per openstack's documentation it is a 'block of IP addresses and associated configuration state. Subnets are used to allocate IP addresses when new ports are created on a network.'
+* network: an isolated Layer 2 networking segment.
+
+### Benefit to oVirt
+
+Using IPv6 addresses is currently only supported in linux-bridge oVirt networks.
+This feature will bring ip addressing type parity to oVirt, thus enabling oVirt
+users to use IPv6 addressing schemes in OVN networks.
+
+## Objectives
+
+Provide guests (virtual machines) attached to OVN networks IPv6 address support.
+The following configurations should be possible:
+
+* static addresses: the user can manually specify the ipv6 address of a port
+* stateful DHCPv6: the user can leverage OVN's native DHCPv6 service to get an ipv6 address in the guests.
+* stateless DHCPv6: the IP address allocated to the guest's interface will be created from the port's MAC address and the IP prefix being announced in the *Router Advertisement* messages. Other connection parameters - like max hops, MTU, dns configs, are set using DHCPv6.
+* stateless address auto-configuration: both the addresses **and** the connection parameters should be automatically configured from RA messages, as per the [SLAAC rfc](https://tools.ietf.org/html/rfc4862).
+
+While the above are all the possible types of configurations, their priority is not the same - the implementation will feature 2 stages:
+
+1. provide support for IPv6 static addresses **and** stateful DHCPv6.
+2. provide support for address auto-configuration from RA messages.
+
+The second stage is **not** mandatory, and will be delivered when possible.
+
+## ovirt-provider-ovn current status & design alternatives
+
+There is a current provider limitation that greatly impacts the design decision
+of this feature: a network can have a **single** subnet defined on top of it.
+
+If this limitation is kept in place, to have VMs connected to both ipv4 and ipv6
+networks, the VMs must be connected to 2 different networks (through 2 different
+ports), each with an address type specific (4 or 6) subnet defined on top.
+
+***QUESTION:*** why is this limitation in place? Why do we enforce a single
+		subnet per network ?
+		My theory is limiting one network to having a single subnet
```

Comment:
@dankenigsberg : can we assume this limitation ?

Reply:
If my memory serves me right, we could not use multiple subnets because ovirt-engine is unaware of them. When it creates a port, it would choose one from a random subnet, so it would randomly be on the same broadcast domain with other ports that belong to the same network.

I think that the same applies to ipv4/ipv6, and I'm fine with documenting and supporting only a single-stack solution.